src/utils/use-beacons.ts

```diff
@@ -0,0 +1,133 @@
+import {useIsBeaconsEnabled} from '@atb/beacons/use-is-beacons-enabled';
+import {KETTLE_API_KEY} from '@env';
+import {useCallback, useEffect, useState} from 'react';
+import {NativeModules, Platform} from 'react-native';
+import {Kettle, KettleModules} from 'react-native-kettle-module';
+import {KettleConsents} from 'react-native-kettle-module';
+import {PERMISSIONS, RESULTS, request} from 'react-native-permissions';
+
+const BEACONS_CONSENTS = [KettleConsents.SURVEYS, KettleConsents.ANALYTICS];
+
+export const useBeacons = () => {
+  const [isBeaconsEnabled] = useIsBeaconsEnabled();
+  const isBeaconsSupported = isBeaconsEnabled && !!KETTLE_API_KEY;
+
+  const [isKettleStarted, setIsKettleStarted] = useState(false);
+  const [kettleIdentifier, setKettleIdentifier] = useState(null);
+  const [kettleConsents, setKettleConsents] = useState();
+  const [isBeaconsOnboarded, setIsBeaconsOnboarded] = useState(false);
+
+  const initializeBeacons = useCallback(async () => {
+    // This will set kettle if KETTLE_API_KEY is set on the native side
+    await NativeModules.KettleSDKExtension.initializeKettleSDK();
+  }, [NativeModules]);
+
+  useEffect(() => {
+    if (isBeaconsSupported) {
+      initializeBeacons();
+    }
+  }, [initializeBeacons, isBeaconsSupported]);
+
+  useEffect(() => {
+    let isMounted = true;
+
+    async function checkKettleInfo() {
+      const status = await Kettle.isStarted();
+      const identifier = await Kettle.getIdentifier();
+      const consents = await Kettle.getGrantedConsents();
+
+      if (isMounted) {
+        setIsKettleStarted(status);
+        setKettleIdentifier(identifier);
+        setKettleConsents(consents);
+        setIsBeaconsOnboarded(Object.keys(consents).length > 0);
```

Comment:
Since these are all set at the same time, could they be a single state variable?